test/ectest.c

```diff
@@ -2170,7 +2119,7 @@ static int cardinality_test(int n)
 
     if (!TEST_ptr(ctx = BN_CTX_new())
         || !TEST_ptr(g1 = EC_GROUP_new_by_curve_name(nid))
-        || !TEST_ptr(g2 = EC_GROUP_new(EC_GROUP_method_of(g1)))) {
+        || !TEST_ptr(g2 = EC_GROUP_new_by_curve_name(nid))) {
```

Comment:
I guess this test could be just dropped now?
I think it was explicitly testing EC_GROUP_method_of, now is not doing much really!

Reply:
I ... think I wrote that line. Further down in the function, it's manipulating `g2` to muck with the parameters, then recalculating them and checking vs `g1`. So `g1` is kind of the "control" `EC_GROUP` for the test.

Reply:
Oh, yeah, I missed how g2 was being used!

Reply:
But yea you're right I was lazy there -- it would be better to not instantiate `g1` and `g2` literally from the same call, to keep them as independent as possible for a valid control. I'll adjust it